src/lib/db.js

```diff
@@ -878,6 +878,28 @@ export function searchStaffQuery(queries, min, max) {
   });
 }
 
+export function searchTagsQuery(queries, min, max) {
+  return new Promise(resolve => {
+    let queriesReturned = 0;
+    const results = {};
+    queries.forEach(query => {
+      database
+        .select('slug')
+        .from('tags')
+        .where('name', 'like', `%${query}%`)
+        .limit(max - min + 1)
+        .offset(min)
+        .then(rows => {
+          queriesReturned += 1;
+          results[query] = _.map(rows, row => row.slug);
+          if (queriesReturned >= queries.length) {
```

Comment:
This is the old ugly way of doing this (that I introduced, so I'll take full blame for that), but let's just use `.map` and `Promise.all` with async / await syntax to write this cleanly, I hate the queriesReturned pattern (which again, I acknowledge that I was the one to introduce :P)